python/galois/lonestar/analytics/average_degree_connectivity.py

```diff
@@ -0,0 +1,142 @@
+import numpy as np
+import numba.types
+import pyarrow
+
+from galois.loops import do_all, do_all_operator
+from galois.property_graph import PropertyGraph
+from galois.shmem import setActiveThreads
+from .average_neighbor_degree import average_neighbor_degree
+
+#SUMMARY OF THE ALGORITHM: 
+##the average degree connectivity of a degree in graph g is the sum 
+#of the average neighbor degree for nodes with degree k divided by 
+#the number of nodes with that degree
+
+#Thus the first step is to create two dictionaries that will hold all
+#of the in/out degrees seen in graph g these are name in_dict and out_dict
+#where key = degree and value = array of node_id's with that degree 
+
+#Second step is to have a method that calculate the sume of the average
+#neighbor degree for nodes with degree k
+
+#Thus when we call get_avg_degconn, we can use our in_dict and out_dict 
+#to go through every degree in the dictionary and calculate the 
+#average degree connectivity for degree k by calling 
+# avg_n_sum / value of key (degree) which the number of nodes with that degree
+#We store this in a result dictionary where
+#key = degree and value = degree's average degree connectivity
+
+
+#QUESTIONS: ask about correct way to pass parameters :) example: line 76
+#TODO: fix indentation
+
+#method to get the sum of the average neighbor degree for nodes with degree k 
```

Comment:
Documentation for how a function is used (so what it's arguments should be and what it returns) should be in a doc string. Please move your function documentation comments into doc strings throughout.

Nit: Also, this is not actually a method. It's a function.